src/puppetlabs/pcp/client.clj

```diff
@@ -142,10 +142,13 @@
   "extracts the common name from the named certificate and forms a PCP
   Uri with it and the supplied type"
   [certificate type]
-  (let [x509     (ssl-utils/pem->cert certificate)
-        cn       (ssl-utils/get-cn-from-x509-certificate x509)
-        identity (format "pcp://%s/%s" cn type)]
-    identity))
+  (let [x509-chain (ssl-utils/pem->certs certificate)]
+    (when (empty? x509-chain)
+      (throw (IllegalArgumentException.
+               (i18n/trs "{0} must contain at least 1 certificate" certificate))))
+    (let [cn       (ssl-utils/get-cn-from-x509-certificate (first x509-chain))
```

Comment:
I've found several sources saying it's ordered, starting with the leaf certificate first - https://support.dnsimple.com/articles/what-is-ssl-certificate-chain/, http://superuser.com/a/347614 - but nothing authoritative.